cloud/types_previews.go

```diff
@@ -0,0 +1,142 @@
+// Copyright 2024 Terramate GmbH
+// SPDX-License-Identifier: MPL-2.0
+
+package cloud
+
+import "github.com/terramate-io/terramate/errors"
+
+type (
+	// PreviewStacks is a list of stack objects for the request payload
+	PreviewStacks []PreviewStack
+
+	// PreviewStack represents the stack object of the request payload
+	// type for the creation of stack deployments.
+	PreviewStack struct {
+		PreviewStatus   string   `json:"preview_status"`
+		Repository      string   `json:"repository"`
+		Path            string   `json:"path"`
+		MetaID          string   `json:"meta_id"`
+		MetaName        string   `json:"meta_name"`
+		MetaDescription string   `json:"meta_description"`
+		MetaTags        []string `json:"meta_tags"`
+		DefaultBranch   string   `json:"default_branch"`
+		Cmd             []string `json:"cmd"`
+	}
+	// CreatePreviewPayloadRequest is the request payload for the creation of
+	// stack deployments.
+	CreatePreviewPayloadRequest struct {
+		UpdatedAt       int64                    `json:"updated_at"`
+		Technology      string                   `json:"technology"`
+		TechnologyLayer string                   `json:"technology_layer"`
+		ReviewRequest   *DeploymentReviewRequest `json:"review_request,omitempty"`
+		Metadata        *DeploymentMetadata      `json:"metadata,omitempty"`
+		Stacks          PreviewStacks            `json:"stacks"`
```

Comment:
There is no usecase for `nil` stacks currently. There is a validation for it in the `Validate()` method in case its ever set as `nil`. The API marks this as a required attribute. Could you please link to the case where it is marked as omitempty? I don't see a problem with marking this as `omitempty` though if it's safer for the API.